packages/server-wallet/src/handlers/__test__/close-channel.test.ts

```diff
@@ -0,0 +1,28 @@
+import matchers from '@pacote/jest-either';
+
+import {closeChannel, CloseChannelError} from '../close-channel';
+import {signStateFixture} from '../../protocols/__test__/fixtures/actions';
+import {channelStateFixture} from '../../protocols/__test__/fixtures/channel-state';
+import {ChannelState} from '../../protocols/state';
+
+expect.extend(matchers);
+
+test('validClose', () => {
+  const cs = channelStateFixture();
+  const {channelId} = cs;
+  expect(closeChannel({channelId}, cs)).toMatchRight(
+    signStateFixture({isFinal: true, turnNum: cs.supported.turnNum + 1})
+  );
+});
+
+const notMyTurnErr = new CloseChannelError(CloseChannelError.reasons.notMyTurn);
+const noSupportedStateErr = new CloseChannelError(CloseChannelError.reasons.noSupportedState);
+
+const noSupportedState: ChannelState = {...channelStateFixture(), supported: undefined};
+test.each`
+  channelState                                      | result
+  ${noSupportedState}                               | ${noSupportedStateErr}
+  ${channelStateFixture({supported: {turnNum: 4}})} | ${notMyTurnErr}
+`('error cases $result', ({channelState, result}) => {
+  expect(closeChannel({channelId: channelState.channelId}, channelState)).toEqualLeft(result);
+});
```

Comment:
Personally I find this a bit hard to read

Reply:
Hmm. Is it just that you're unfamiliar with it?

To me, it is vastly easier to understand the test cases when organized in this fashion. Refactoring the application test cases highlighted that for me. You could instead group together similar tests in nested describe blocks, but then you have to look at many different places to understand the test setup

Reply:
I agree it's easier to read test cases in a tabular manner like this, my initial difficulty in reading it was moreso todo with formatting. For example, maybe instead of `channelStateFixture({supported: {turnNum: 4}})` you could label this variable as `evenTurnNumState`. There could also be a description column, so that when I run the tests I see an english description of exactly what is being tested.